File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/StrategyPolicyMetaPO.java
package io.iamcyw.ams.job.strategy.persistence;

import javax.persistence.*;

@Entity(name = "ALARM_STRATEGY_POLICY_META")
public class StrategyPolicyMetaPO {
    @Id
    @GeneratedValue
    public Long id;

    public MessageScope scope;

    public String contrastKey;

    public String contrastValue;

    public PolicyContrastType contrastType;

    @ManyToOne(fetch = FetchType.LAZY)
    public StrategyPolicyPO policy;

    public enum MessageScope {
        Headers, Payload
    }

    public enum PolicyContrastType {
        Fixed, Express, Pass
    }

}

File: domain/src/commonMain/kotlin/io/iamcyw/ams/domain/job/strategy/StrategyPolicyMetaPO.kt
package io.iamcyw.ams.domain.job.strategy

expect open class StrategyPolicyMetaPO {

    var scope: MessageScope?

    var contrastKey: String

    var contrastValue: String

    var contrastType: PolicyContrastType?


}

enum class MessageScope {
    Headers, Payload
}

enum class PolicyContrastType {
    Fixed, Express, Pass
}

